airflow/executors/base_executor.py

```diff
@@ -130,6 +131,10 @@ def __init__(self, parallelism: int = PARALLELISM):
         self.running: set[TaskInstanceKey] = set()
         self.event_buffer: dict[TaskInstanceKey, EventBufferValueType] = {}
         self.attempts: dict[TaskInstanceKey, RunningRetryAttemptType] = defaultdict(RunningRetryAttemptType)
+        self.task_context_logger: TaskContextLogger = TaskContextLogger(
+            component_name="Executor",
```

Comment:
Should each executor create one of these loggers? In a world of hybrid executor you may have multiple and this name will be shared amongst them?

Reply:
I hesitated. I dont see a concern having one logger shared among multiple executors? If you prefer having multiple one, I can also do it. I dont feel very opinionated on this one

Reply:
I'm fine either way, I'm not really sure what all the implications of sharing that logger are. So if you've tested it and it's working well then no worries!

Reply:
i would make the name lowercase because this will end up in the filename.  maybe update TaskContextLogger to raise an error if given a non-lowercase component name?